### PR TITLE
Mon 2771 change templates to select2

### PR DESCRIPTION
**Pull request by miteto**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -24,7 +20,7 @@
         "phing/phing": "^2.16",
         "behat/mink-extension": "^2.2",
         "behat/mink-selenium2-driver": "^1.3",
-        "centreon/centreon-test-lib": "dev-master",
+        "centreon/centreon-test-lib": "dev-MON-2771-change-templates-to-select2",
```
Comment (sc979) [inline composer.json]: to remove before merging

File: www/api/class/centreon_configuration_hosttemplate.class.php [hunk 2]
```diff
@@ -52,6 +52,36 @@ public function __construct()
...
+        }
+
+        $result = [];
+        $query = 'SELECT htr.host_tpl_id AS id, h.host_name
+                  FROM host_template_relation htr
+                  JOIN host h ON htr.host_tpl_id = h.host_id
+                  WHERE htr.host_host_id = :host_id
+                  ORDER BY htr.`order`';
+        $stmt = $this->pearDB->prepare($query);
+        $stmt->bindParam(':host_id', $id);
```
Comment (sc979) [inline www/api/class/centreon_configuration_hosttemplate.class.php]: please use bindValue

File: www/include/configuration/configObject/host/formHost.php [hunk 3]
```diff
@@ -323,7 +308,7 @@ function allInSameInstance($hosts, $instanceId)
     'multiple' => true,
     'linkedObject' => 'centreonHost'
 );
-$hostTplRoute = './include/common/webServices/rest/internal.php?object=centreon_configuration_hosttemplates'
+$hostTplRoute = './include/common/webServices/rest/internal.php?object=centreon_configuration_hosttemplate&action=list'
```
Comment (sc979) [inline www/include/configuration/configObject/host/formHost.php]: you ask twice the action (line 312)

File: www/include/configuration/configObject/host/formHost.php [hunk 4]
```diff
@@ -445,6 +430,11 @@ function allInSameInstance($hosts, $instanceId)
 );
 $form->addElement('static', 'tplText', _("Using a Template allows you to have multi-level Template connection"));
 
+$tplDeRoute = './include/common/webServices/rest/internal.php?object=centreon_configuration_hosttemplate'
```
Comment (sc979) [inline www/include/configuration/configObject/host/formHost.php]: what means $tplDeRoute ? -> can you choose a more explicit variable name
caution -> change line 435 too

File: www/include/configuration/configObject/host/formHost.php [hunk 5]
```diff
@@ -1251,9 +1230,8 @@ function myReplace()
     $tpl->assign('template_inheritance', _('Template inheritance'));
     $tpl->assign('command_inheritance', _('Command inheritance'));
     $tpl->assign('cloneSetMacro', $cloneSetMacro);
-    $tpl->assign('cloneSetTemplate', $cloneSetTemplate);
     $tpl->assign('centreon_path', $centreon->optGen['oreon_path']);
-    $tpl->assign("k", $k);
+    //$tpl->assign("k", $k);
```
Comment (sc979) [inline www/include/configuration/configObject/host/formHost.php]: please remove dead code

File: www/include/configuration/configObject/host/formHostTemplate.ihtml [hunk 6]
```diff
@@ -0,0 +1,635 @@
...
+<form {$form.attributes}>
+     <div  class="headerTabContainer">
+        <ul id="mainnav">
+            <li class="a" id='c1'><a href="#"  style='cursor:pointer' onclick="javascript:montre('1');">{$sort1}</a></li>
+            <li class="b" id='c2'><a href="#" style='cursor:pointer' onclick="javascript:montre('2');">{$sort2}</a></li>
+            <li class="b" id='c3'><a href="#" style='cursor:pointer' onclick="javascript:montre('3');">{$sort3}</a></li>
+            <li class="b" id='c4'><a href="#" style='cursor:pointer' onclick="javascript:montre('4');">{$sort4}</a></li>
+            <li class="b" id='c5'><a href="#" style='cursor:pointer' onclick="javascript:montre('5');">{$sort5}</a></li>
+        </ul>
+		<div id="validFormTop">
```
Comment (sc979) [inline www/include/configuration/configObject/host/formHostTemplate.ihtml]: odd indentation

File: www/include/configuration/configObject/host/formHostTemplate.ihtml [hunk 7]
```diff
@@ -0,0 +1,635 @@
...
+	        </tr>
+ 	        <tr class="list_lvl_1">
+ 	            <td class="ListColLvl1_name" colspan="2"><h4>{t}Host basic information{/t}</h4></td>
+ 	        </tr>
+	    	{if $o != "mc"}
+		    <tr class="list_one">
+		        <td class="FormRowField"><img class="helpTooltip" name="host_name">{$form.host_name.label}</td>
+		        <td class="FormRowValue">{$form.host_name.html}</td>
+		    </tr>
+			<tr class="list_two">
```
Comment (sc979) [inline www/include/configuration/configObject/host/formHostTemplate.ihtml]: please recheck all this file indentation

File: www/include/configuration/configObject/host/formHost.php (hunk 4 above)
Comment (miteto) [reply www/include/configuration/configObject/host/formHost.php]: I saw "$timezDeRoute" at line 391 so I tried to be as consistent as possible. Will change it.

File: composer.json (hunk 1 above)
Comment (sc979) [reply composer.json]: if you've already builded your branch on jenkins and acceptance tests passed, you don't need this anymore. Please remove it...

**Pull request closed without merging**